src/main/scala/firrtl/Visitor.scala

```diff
@@ -105,14 +105,29 @@ class Visitor(infoMode: InfoMode) extends FIRRTLBaseVisitor[FirrtlNode] {
         if (ctx.moduleBlock() != null)
           visitBlock(ctx.moduleBlock())
         else EmptyStmt)
-      case "extmodule" => ExtModule(info, ctx.id.getText, ctx.port.map(visitPort))
+      case "extmodule" =>
+        val defname = if (ctx.defname != null) ctx.defname.id.getText else ctx.id.getText
+        val ports = ctx.port map visitPort
+        val params = ctx.parameter map visitParameter
+        ExtModule(info, ctx.id.getText, ports, defname, params)
     }
   }
 
   private def visitPort[FirrtlNode](ctx: FIRRTLParser.PortContext): Port = {
     Port(visitInfo(Option(ctx.info), ctx), ctx.id.getText, visitDir(ctx.dir), visitType(ctx.`type`))
   }
 
+  private def visitParameter[FirrtlNode](ctx: FIRRTLParser.ParameterContext): Param = {
```

Comment:
In general, is the point of visiting to go through the input *.fir file, match to the FIRRTL.g4 "template" and then parse parameters via the below match? And therefore you can't do anything else except have all of these nulls?